pulsar-common/src/main/java/org/apache/pulsar/common/protocol/schema/SchemaHash.java

```diff
@@ -60,7 +68,8 @@ public static SchemaHash of(SchemaInfo schemaInfo) {
     }
 
     public static SchemaHash of(byte[] schemaBytes, SchemaType schemaType) {
-        return new SchemaHash(hashFunction.hashBytes(schemaBytes), schemaType);
+        return cache.get(Pair.of(schemaBytes, schemaType),
+                schemaTypePair -> new SchemaHash(hashFunction.hashBytes(schemaBytes), schemaType));
```

Comment:
This would creating a capturing lambda each time. We could optimize by creating a `LoadingCache`, with its loading method.